src/main/java/net/atos/zac/shared/helper/SuspensionZaakHelper.java

```diff
@@ -76,40 +78,55 @@ public Zaak hervattenZaak(final Zaak zaak, final String redenHervatting) {
         }
         final LocalDate uiterlijkeEinddatumAfdoening = zaak.getUiterlijkeEinddatumAfdoening().plusDays(offset);
 
-        final String toelichting = String.format("%s: %s", HERVATTING, redenHervatting);
-        final Zaak updatedZaak = zrcClientService.patchZaak(
-                zaakUUID,
-                toPatch(einddatumGepland, uiterlijkeEinddatumAfdoening, redenHervatting, false),
-                toelichting
+        final String toelichting = String.format("%s: %s", RESUMING, resumeReason);
+        final var patchZaak = addSuspensionToZaakPatch(
+                createZaakPatch(einddatumGepland, uiterlijkeEinddatumAfdoening),
+                resumeReason,
+                false
         );
+
+        final Zaak updatedZaak = zrcClientService.patchZaak(zaakUUID, patchZaak, toelichting);
         zaakVariabelenService.removeDatumtijdOpgeschort(zaakUUID);
         zaakVariabelenService.removeVerwachteDagenOpgeschort(zaakUUID);
         return updatedZaak;
     }
 
-    /**
-     * @param einddatumGepland             streefdatum van de zaak; may be null in which case the
-     *                                     streefdatum is not patched
-     * @param uiterlijkeEinddatumAfdoening fataledatum van de zaak
-     * @param reden                        reden voor de opschorting
-     * @param isOpschorting                true indien opschorten, false indien hervatten
-     * @return zaak voor patch
-     */
-    private Zaak toPatch(
-            final LocalDate einddatumGepland,
-            final LocalDate uiterlijkeEinddatumAfdoening,
-            final String reden,
-            final boolean isOpschorting
+    public Zaak extendZaakFatalDate(final Zaak zaak, final long numberOfDays, final String description) {
```

Comment:
I am a little bit surprised that extend is a calculated numberOfDays; but that must be the common pattern in this part of the codebase.

Reply:
I'm reusing the existing logic to minimize the duplicating code, hence the `numberOfDays` here. We're also changing the `EinddatumGepland` with the same number of days and this might not be the same date as fatal one.